Add CheckAddrWithOptions method with IPv4/IPv6

Lan-ce-lot opened the pull request:
### Does this pull request fix one issue?
Fix #30 

### Describe how you did it
- Created CheckAddrWithOptions(addr, opts) method
- Enhanced [parseSockAddr] to support network parameter (tcp, tcp4, tcp6)
- ~~Added SO_MARK socket option support for Linux~~

### Describe how to verify it
```
// IPv4 only
opts := tcp.DefaultOptions().WithNetwork("tcp4")
checker.CheckAddrWithOptions("google.com:80", opts)

// IPv6 only
opts := tcp.DefaultOptions().WithNetwork("tcp6") 
checker.CheckAddrWithOptions("google.com:80", opts)

// Run tests
go test -v ./...
```
<img width="925" height="867" alt="Clipboard_Screenshot_1757909672" src="https://github.com/user-attachments/assets/b39400ed-40b4-4185-a428-cc792d279d6c" />

Lan-ce-lot committed:
Add CheckAddrWithOptions method with IPv4/IPv6 and socket mark support

Lan-ce-lot commented:
I passed the CI in my own repo. PTAL @tevino

tevino commented:
@Lan-ce-lot Thanks for your work, I'll be mostly AFK for the next week or so, but I’ll definitely check in after that!

Lan-ce-lot commented on socket_linux.go:
```diff
@@ -12,15 +12,28 @@ import (
+
+	if zeroLinger {
+		if err = _setZeroLinger(fd); err != nil {
+			_ = unix.Close(fd)
```
Thanks!

Lan-ce-lot commented on socket_unix.go:
```diff
@@ -10,7 +10,12 @@ import (
 
 // parseSockAddr resolves given addr to unix.Sockaddr
 func parseSockAddr(addr string) (sAddr unix.Sockaddr, family int, err error) {
```
Done.

Lan-ce-lot commented on test_server_test.go:
```diff
@@ -12,3 +13,29 @@ func StartTestServer() (string, context.CancelFunc) {
 }
+
+// StartTestServerIPv6 starts a test HTTP server on IPv6 loopback and returns its address and a cancel function
+func StartTestServerIPv6() (string, context.CancelFunc, error) {
```
Done. StartTestServer now uses an explicit tcp4 listener on 127.0.0.1:0, so it won't fall back to IPv6. I kept the IPv6 helper separate for the IPv6-specific tests.

Lan-ce-lot commented on options_linux.go:
```diff
@@ -0,0 +1,62 @@
+	// This forces the connection to be reset immediately when closed
+	ZeroLinger bool
+
+	// Mark sets the SO_MARK socket option (Linux only)
```
Done. I removed it.

Commit:
Align naming with implementation

Commit:
Relocate Checker integration tests from options_test to checker_test

Commit:
Log test-server bind addresses and test IPv6 address in assertions

Commit:
Clarify that "tcp" network has no IPv4/IPv6 fallback

The old comment said '"tcp" - Try IPv4 first, then IPv6 (default behavior)'
but the implementation only resolves a single address via net.ResolveTCPAddr
and makes one connection attempt. There is no iteration or fallback between
address families.

Commit:
Add @Lan-ce-lot to contributors

tevino commented on options.go:
```diff
@@ -0,0 +1,51 @@
+
+	// Network specifies the network type for address resolution and connection
+	// Supported values:
+	//   "tcp"  - Try IPv4 first, then IPv6 (default behavior)
```
> ”tcp“ - Try IPv4 first, then IPv6 (default behavior)

This line originates from #30 which differs from the implementation of this PR.

I find the approach better: to avoid handling the complexities in this package and retain that flexibility for the caller.

tevino commented:
@Lan-ce-lot Thanks for your contribution!
